src/Expectation.php

```diff
@@ -324,6 +325,41 @@ public function when(callable|bool $condition, callable $callback): self
         return $this;
     }
 
+    /**
+     * Asserts that only one of the given tests pass with the given expectation target.
+     *
+     * @param  (\Closure(self<TValue>): (mixed|void))  ...$tests
+     * @return self<TValue>
+     */
+    public function toBeOneOf(Closure ...$tests): self
+    {
+        if ($tests === []) {
+            return $this;
+        }
+
+        $matches = [];
+        $exceptions = [];
+
+        foreach ($tests as $key => $test) {
+            try {
+                $test(new Expectation($this->value));
+                $matches[] = $key;
+            } catch (AssertionFailedError) {
+                $exceptions[] = $key;
+            }
+        }
+
+        if (count($matches) === 1) {
+            return $this;
+        }
+
+        if (count($matches) > 1) {
+            throw new ExpectationFailedException('Failed asserting value matches exactly one expectation (matches: '.implode(', ', $matches).').');
```

Comment:
Rather than outputting the keys, I'd just output a count of passes and fails for a few reasons:

- Keys are limited to ints, or variable names due to using the splat operator rather than accepting an iterable, for example: `->toBeOneOf(thisIsTheOnlyWayToNameAnExpectation: fn($e) => ..., spaces_are_not_allowed: fn($e) => ...)`
- If you were accepting an iterable rather than using a splat array keys aren't guaranteed to be unique
- There could be a ton of expectations passed to this function, causing a failure to output a huge error

You can output the count instead by by checking the delta of `Assert::getCount()` before and after and by counting the exceptions rather than capturing them in an array. This has the added benefit of reducing the memory overhead.